### PR TITLE
Verify osd tree output and ensure it have device set PVC names

### PR DESCRIPTION
**Pull request by jilju**
Updated ocs_install_verification function to verify ceph osd tree output.

ceph osd tree output structure and values are verified.
Using JSON Schema to validate the output.
Presence of device set PVC names is also verified along with this. (Verifies bug https://bugzilla.redhat.com/show_bug.cgi?id=1765154)
Implemented in this way because there are more checks to be done
in the output. Added as TODO.

Signed-off-by: Jilju Joy <jijoy@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/resources/ocs.py [hunk 1]
```diff
@@ -349,3 +352,32 @@ def ocs_install_verification(timeout=600, skip_osd_distribution_check=False):
     assert {defaults.CEPHFS_PROVISIONER, defaults.RBD_PROVISIONER} == (
         {item['metadata']['name'] for item in csi_driver.get()['items']}
     )
+
+    # Verify ceph osd tree have device set PVC names specified
+    log.info("Checking for device set PVC names in ceph osd tree output.")
+    deviceset_pvcs = [
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/ocs.py]: please use ocs_ci.ocs.resources.pvc.get_deviceset_pvs

File: ocs_ci/ocs/resources/ocs.py [hunk 2]
```diff
@@ -349,3 +352,32 @@ def ocs_install_verification(timeout=600, skip_osd_distribution_check=False):
...
+    for pvc in deviceset_pvcs:
+        assert f'host {pvc}' in osd_tree, (
+            f"ceph osd tree output does not contain pvc name {pvc}"
+        )
+    log.info("Verified device set PVC names in ceph osd tree output.")
+
+    # Verify OSDs are listed as sdd
+    log.info("Verify OSDs are listed as sdd in ceph osd tree output.")
+    for num in range(len(deviceset_pvcs)):
+        re_exp = f'ssd [0-9]*.[0-9]*[ ]*osd.{num}'
```
Comment (vasukulkarni) [inline ocs_ci/ocs/resources/ocs.py]: this is dependent on hardware we run and can easily fail if we move out of aws or different systems?

**Comment by jilju** [comment]
circular dependencies to be resolved

File: ocs_ci/ocs/resources/ocs.py (hunk 1 above)
Comment (jilju) [reply ocs_ci/ocs/resources/ocs.py]: get_deviceset_pvs is giving list of PV. Names of PVCs are required here.

File: ocs_ci/ocs/resources/ocs.py (hunk 2 above)
Comment (jilju) [reply ocs_ci/ocs/resources/ocs.py]: This part is removed now. It will be added in another PR after getting this clarity.

**Comment by jilju** [comment]
> circular dependencies to be resolved

resolved

File: ocs_ci/ocs/resources/ocs.py (hunk 1 above)
Comment (ebenahar) [reply ocs_ci/ocs/resources/ocs.py]: got you. Maybe would be good to create a function for this implementation in pvc.py

**Comment by vasukulkarni** [comment]
I see couple for changes and its worth testing again.

File: ocs_ci/ocs/resources/ocs.py (hunk 1 above)
Comment (jilju) [reply ocs_ci/ocs/resources/ocs.py]: Added new function get_deviceset_pvcs in pvc.py

File: ocs_ci/ocs/resources/ocs.py [hunk 3]
```diff
@@ -363,3 +366,32 @@ def ocs_install_verification(timeout=600, skip_osd_distribution_check=False):
...
+    for item in osd_tree['nodes']:
+        validate(instance=item, schema=schemas[item['type']])
+        if item['type'] == 'host':
+            deviceset_pvcs.remove(item['name'])
+    log.info(
+        "Verified ceph osd tree output. Device set PVC names are given in the "
+        "output."
+    )
+
+    # TODO: Verify ceph osd tree output have osd listed as ssd
```
Comment (jilju) [inline ocs_ci/ocs/resources/ocs.py]: Verifies bug https://bugzilla.redhat.com/show_bug.cgi?id=1770910

**Pull request merged**